### PR TITLE
feat(infra): migrate to lerna 8

### PR DESCRIPTION
**Pull request by BatuhanW**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://refine.dev/docs/guides-concepts/contributing/#commit-convention

## Bugs / Features

- [ ] Docs have been added / updated
- [x] Changesets have been added https://refine.dev/docs/guides-concepts/contributing/#creating-a-changeset

## What is the current behavior?

Currently, we are using `lerna bootstrap`.

## What is the new behavior?

- Migrated to `pnpm workspaces`
- Updated lerna to 8
- Updated NX config
- Removed leftover `gitHead` field from package.json's
- Disabled telemetry for e2e tests.

fixes # (issue)

### CODE REVIEW

File: documentation/docs/guides-concepts/contributing/index.md [hunk 1]
```diff
@@ -37,101 +37,55 @@ After you fork the Refine repository, you need to clone it to your local machine
 git clone https://github.com/refinedev/refine.git
 ```
 
-### Installing dependencies
+### Installing root dependencies
```
Comment (aliemir) [inline documentation/docs/guides-concepts/contributing/index.md:40]:
```suggestion
### Installing dependencies
```

I think we don't need to say root dependencies anymore

File: documentation/docs/guides-concepts/contributing/index.md [hunk 2]
```diff
@@ -37,101 +37,55 @@ After you fork the Refine repository, you need to clone it to your local machine
...
-
-- We're using `--scope` flag to filter the packages and examples we want to bootstrap. In this example, we're bootstrapping `@refinedev/antd` package and `base-antd` example.
-- `--includeDependencies` flag is used to tell Lerna to also include the dependencies of the packages and examples we've filtered to the bootstrapping process.
-
-### Building the Packages
-
-After bootstrapping the packages we want to work on, we need to build them in order to run properly. You can use the command below to build the packages:
-
-```sh title="Terminal"
-npm run build -- --scope @refinedev/antd --scope base-antd --includeDependencies
```
Comment (aliemir) [inline documentation/docs/guides-concepts/contributing/index.md:96]: I think this build part is still relevant. We can remove things like `--includeDependencies` etc. but should still keep the `pnpm build` command referencer somewhere in this document.

File: documentation/docs/guides-concepts/contributing/index.md [hunk 3]
```diff
@@ -147,17 +101,17 @@ Refine documentation is built with [Docusaurus](https://docusaurus.io/). Documen
 
 ```sh title="Terminal"
 cd documentation
-npm install
-npm run dev:docs
+pnpm --ignore-workspace install
```
Comment (aliemir) [inline documentation/docs/guides-concepts/contributing/index.md:104]: Now we've defined `/documentation` as a separate workspace and don't need to use `--ignore-workspace` anymore.

**Review by aliemir — APPROVED** [review]
🚀

**Review by omeraplak — APPROVED** [review]

**Pull request merged**